extends Panel

signal showItem

@onready
var message = $message

var active: bool = false

# Called when the node enters the scene tree for the first time.
func _ready():
	visible = false

var yuhina_and_god = [
	"[b][color=blue]ç¥æ§˜[/color][/b] ã‚„ã‚ã‚ˆãæ¥ãŸãªã€yuhina!",
	"[b][color=red]yuhina[/color][/b] ã‚ï¼ãã®å£°ã¯3å¹´å‰ã«æ­»ã‚“ã ã¯ãšã®ç¥æ§˜ã§ã¯â€¼ï¸",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã»ã»ã»ã»ã€é©šã„ãŸã‹ğŸ‘»",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã¡ãªã¿ã«ã“ã‚“ãªé•·ã„ã‚»ãƒªãƒ•ã ã£ã¦é•·ã•ã‚’æ°—ã«ã›ãšä¸€æ°—ã«å–‹ã‚Œã‚‹ã®ã¯Scratchã«ã¯ç„¡ç†ãªèŠ¸å½“ã ã‚ã†ã€‚ã©ã†ã ï¼",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã€ã€ã€ã¾ã‚ã€é™ç•Œã¯ã‚ã‚‹ãŒãªã€‚ãã‚Œã«ãƒœã‚¿ãƒ³ã‚’æŠ¼ã•ãªãã¦ã‚‚å‹æ‰‹ã«é€²ã‚€ãŒãªğŸ˜€",
	"[b][color=red]yuhina[/color][/b] ã§ã€ã“ã‚“ãªã¨ã“ã‚ã§ä½•ã‚’ã—ã¦ã„ã‚‹ã®ã§ã™ã‹ï¼ï¼Ÿ",
	"[b][color=blue]ç¥æ§˜[/color][/b] ãŠå‰ã«ä¼èª¬ã®ã‚¢ã‚¤ãƒ†ãƒ ã‚’æ¸¡ã™ãŸã‚ã˜ã‚ƒã€ãã‚Œï¼ğŸ",
]

var yuhina_and_god2 = [
	"[b][color=blue]ç¥æ§˜[/color][/b] ã‚‚ã†ãŠå‰ã«å–‹ã‚‹ã“ã¨ã¯ãªã„ãª",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã‚¤ãƒ™ãƒ³ãƒˆã‚’ã‚¯ãƒªã‚¢ã—ã¦ã‹ã‚‰ã¾ãŸæ¥ã‚‹ãŒã‚ˆã„ã€‚"
]

func _on_god_yuhina_in(first):
	if first:
		start_conversation(yuhina_and_god, true)
	else:
		start_conversation(yuhina_and_god2, false)

func start_conversation(lines, show_item):
	visible = true
	for l in lines:
		message.text = l
		$Timer.start()
		await $Timer.timeout
	visible = false
	if show_item:
		showItem.emit()
